### PR TITLE
Split up bind functions and refactor tests

### PR DESCRIPTION
**Pull request by hadley**
Test coverage stayed the constant. But I had to make more value judgements here — I remove the tests for vector coercion edge cases (figuring that's vctrs job) but kept the big picture tests. I also kept all the tests that pertain to the magic the `bind_rows()` adds on top of `vec_rbind()`.

### CODE REVIEW

**Comment by hadley** [comment]
I added some mild `bind_rows()` refactoring because I couldn't help myself. I'm fairly confident I kept the logic the same.

**Review by DavisVaughan — APPROVED** [review]
A bit hard to tell which tests got deleted, but the ones you left look good and I trust you

File: R/bind-rows.R [hunk 1]
```diff
@@ -36,6 +36,12 @@ bind_rows <- function(..., .id = NULL) {
   }
```
[the remark this answers is not in the document]
Comment (hadley) [reply R/bind-rows.R:36]: I think that's relatively low risk

File: R/bind-rows.R [hunk 2]
```diff
@@ -61,30 +65,18 @@ bind_rows <- function(..., .id = NULL) {
...
-  dots <- map(dots, function(.x) {
-    if (vec_is_list(.x)) {
-      .x <- vctrs::data_frame(!!!.x, .name_repair = "minimal")
-    }
-    .x
-  })
-
-  if (is.null(.id)) {
+  } else {
+    # Needed to avoid some vctrs error I don't understand
```
[the remark this answers is not in the document]
Comment (hadley) [reply R/bind-rows.R:69]: Yeah, thanks!
